src/grpc/infer_handler.h

```diff
@@ -767,12 +806,25 @@ class InferHandlerState {
         }
       }
 
-      LOG_VERBOSE(1) << "Completing cancellation for " << name
-                     << ", rpc_ok=" << rpc_ok << ", context "
-                     << state->context_->unique_id_ << ", " << state->unique_id_
-                     << " step " << state->step_;
+      if (state->step_ != Steps::CANCELLATION_ISSUED) {
+        // The cancellation has not been issued hence the state can
+        // be released.
+        LOG_VERBOSE(1) << "Completing cancellation for " << name
+                       << ", rpc_ok=" << rpc_ok << ", context "
+                       << state->context_->unique_id_ << ", "
+                       << state->unique_id_ << " step " << state->step_;
 
-      return false;
+        return false;
+      } else {
+        // Should wait for the ResponseComplete callbacks to be invoked.
+        LOG_VERBOSE(1)
+            << "Waiting for the callback to retrieve cancellation for " << name
+            << ", rpc_ok=" << rpc_ok << ", context "
+            << state->context_->unique_id_ << ", " << state->unique_id_
+            << " step " << state->step_;
```

Comment:
Use new `DebugString()` ?

Reply:
DebugString is too much information and is being printed in super-verbose mode. 
Note that it prints all the states associated with the context and their each steps.